store-and-retrieve-data/use-storagehub-sdk/create-a-bucket.md

```diff
@@ -74,120 +39,155 @@ async function run() {
-```
+Fetch the `valueProps` from the MSP you are connected to. An MSP's value prop is its storage fee. The files you will store within a certain bucket will cost you based on the value prop you choose.
 
-Now that you have the `StorageHubClient` initialized, you'll use it to derive the bucket ID and create the bucket.
+To fetch `valueProps` from the MSP Client, take the following steps:
 
-## Derive Bucket ID
+1. Add the following helper function to your `mspService.ts` file:
 
-Before creating a new bucket, you'll need to derive the bucket ID by passing the bucket's name and the address you intend to use to create it. Two buckets with the same name and address cannot co-exist within the same MSP.
+    ```ts title="mspService.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/msp-service-with-value-props.ts:msp-value-props'
+    ```
 
-```ts title="// **PLACEHOLDER FOR STEP 3: DERIVE BUCKET ID**"
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/create-a-bucket.ts:derive-bucket'
-```
+2. Add the `getValueProps` method to the export statement at the bottom of the `mspService.ts` file.
 
-Run the script:
+    ```ts title="mspService.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/msp-service-with-value-props.ts:exports'
+    ```
 
-```bash
-ts-node index.ts
-```
+??? code "View complete `mspService.ts` file"
 
-The response should include something like this:
+    ```ts title="mspService.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/msp-service-with-value-props.ts'
+    ```
 
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/output-02.html'
+### Add Method to Create a Bucket
 
-## Check If Derived Bucket ID Isn’t Already On-Chain
+Bucket-related logic will live in a separate `bucketOperations.ts` file. To implement it, take the following steps:
 
-Now that you have the bucket ID, you can ensure the bucket doesn't exist on-chain yet. 
+1. Create a new folder called `operations` within the `src` folder (at the same level as the `services` folder) like so:
 
-```ts title="// **PLACEHOLDER FOR STEP 4: CHECK IF BUCKET EXISTS**"
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/create-a-bucket.ts:check-bucket'
-```
+    ```bash
+    mkdir operations
+    ```
 
-!!! note
-    The `mspClient` can also be used to check if the derived bucket ID already exists using the `mspClient.buckets.getBucket()` function; however, this only checks if that specific MSP contains a bucket with that ID.
+2. Create a new file within the `operations` folder called `bucketOperations.ts`
 
-If you rerun the script, the response should include:
+3. Add the following code:
 
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/output-03.html'
+    !!! note
+        Within the `createBucket` helper method, the `getValueProps` helper method from the previous section is used.
```

Comment:
The number of admonitions on a page can get out of hand quickly and when a page has a ton of admonitions, it adds visual clutter and reduces effectiveness. So I'd probably instead do something like:

3. Add the following code, which uses the `getValueProps` helper from the previous section in `createBucket`: